tools/fastq_provider/fastq_provider.xml

```diff
@@ -0,0 +1,89 @@
+            <option value="auto" selected="true">auto</option>
+            <option value="ena">ENA</option>
+            <option value="sra">SRA file</option>
+            <option value="hca">Human Cell Atlas</option>
+            <option value="dir">Local directory</option>
+	    </param>
+        <param name="retrieval_method" argument="-m" type="text"  label="retrieval method, default 'wget' " value='wget' help="retrieval method, default 'auto'. Files with hca source will use the 'hca' download method regardless (-m has no effect). Private ENA files will be fetched via 'ssh' (-m has no effect). The 'http' method refers to an HTTP endpoint in the EBI Fire resource. "  optional='true'>
+            <option value="auto" selected="true">auto</option>
+            <option value="wget">wget</option>
+            <option value="dir">Local directory</option>
+            <option value="ssh">SSH (ENA files, for EBI only)</option>
+            <option value="http">HTTP endpoint (ENA files, EBI only)</option>
+        </param>
+        <param name="mode" argument="-p" type="select" format="select" label="public or private, default public. Private usable by EBI staff only" help="public or private, default 'public' " optional='true'>
+            <option value="public" selected="true">public</option>
+            <option value="private">private</option>
+	    </param>
+        <param name="library" argument="-l" type="text"  label="ENA library (ENA source files only), by default inferred from file name" help="library, by default inferred from file name. E.g. -l ERR1888646 " optional='true'/>
+        <param name="config_file" argument="-c" type="data" format="atlas-fastq-provider-config.sh" label="config file to override defaults" help="config file to override defaults" optional='true'/>
+        <param name="validate_only" argument="-v" type="boolean" label="validate only to check a source URI is valid, don't download" help="validate only, don't download" optional='true'/>
+        <param name="download_type" argument="-d" type="select" label="download type, fastq or SRA file" help="download type, fastq or srr" optional='true'>
+            <option value="fastq" selected="true">fastq</option>
+            <option value="srr">SRA</option>
+	    </param>
+    </inputs>
+    <outputs>
+        <data name="outfile" label="${tool.name} on ${on_string}: compressed fastq file" format="fastq.gz"/>
+    </outputs>
+    <tests>
+        <test expect_num_outputs="1">
+            <param name="source" value="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR035/ERR035229/ERR035229.fastq.gz"/>
+            <param name="target" value="ERR035229.fastq.gz"/>
+            <output name="outfile"  ftype="fastq.gz" md5="d41d8cd98f00b204e9800998ecf8427e"/>
```

Comment:
`d41d8cd98f00b204e9800998ecf8427e` is the hash of an empty string (I only know this because of recent failures).  So the test is passing with the assumption that you really want an empty output file.  I think planemo/galaxy doesn't know which file to check because there's no 'instruction' on how to identify `outfile` either in the command or in the `outputs` section.  That, or the command is really producing an empty file.

Reply:
thanks @irisdianauy that was very helpful.  And thanks @pcm32 for the suggestion to collected the output here [9be4fb8](https://github.com/ebi-gene-expression-group/container-galaxy-sc-tertiary/pull/258/commits/9be4fb8e8a116660c142f5c66490fd7e69d0c3e0)